### PR TITLE
OCM-3005: Fix allow private cluster without private link

### PR DESCRIPTION
**Pull request by nirarg**
Change the terraform provider logic to align with the logic in the backend

The following cases are allowed:
1. private cluster with private link
2. public cluster without private link
3. private cluster without private link

Public cluster with private link is not allowed

### CODE REVIEW

**Comment by arendej** [comment]
Critical correction:

The following cases are allowed:

private cluster with private link
public cluster without private link
Public cluster with private link
Private cluster without private link is not allowed

File: subsystem/cluster_resource_rosa_test.go [hunk 1]
```diff
@@ -2026,7 +2035,7 @@ var _ = Describe("rhcs_cluster_rosa_classic - create", func() {
 		`)
 		Expect(terraform.Apply()).To(BeZero())
 	})
-	It("Creates cluster with aws subnet ids & private & private link", func() {
+	It("Creates private cluster with aws subnet ids & private link", func() {
```
Comment (bardielle) [inline subsystem/cluster_resource_rosa_test.go:2038]: I think we missed here the test (that should be failed) not Private with private link

**Comment by jharrington22** [comment]
> Critical correction:
> 
> The following cases are allowed:
> 
> private cluster with private link public cluster without private link Public cluster with private link Private cluster without private link is not allowed

We do not support a  "Public cluster with private link"

This PR looks good to me.

**Pull request merged**